configurator/src/main/java/org/apereo/openequella/adminconsole/configurator/ProxySettingsDialog.java

```diff
@@ -0,0 +1,233 @@
+    all.setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+
+    all.add(hostPanel, new Rectangle(0, 0, 3, 1));
+    all.add(credentials, new Rectangle(0, 1, 3, 1));
+    all.add(ok, new Rectangle(1, 2, 1, 1));
+    all.add(cancel, new Rectangle(2, 2, 1, 1));
+
+    updateButtons();
+
+    setTitle("Proxy Settings");
+    setModal(true);
+    setResizable(false);
+    getContentPane().add(all);
+    getRootPane().setDefaultButton(ok);
+
+    pack();
+    ComponentHelper.centreOnScreen(this);
+  }
+
+  protected JPanel createHostPanel() {
+    JLabel hostLabel = new JLabel("Proxy Host:");
+    JLabel portLabel = new JLabel("Proxy Port:");
+
+    hostField = new JTextField();
+    hostField.addKeyListener(this);
+
+    portModel = new SpinnerNumberModel(PROXY_DEFAULT, PROXY_MIN, PROXY_MAX, PROXY_STEP);
+    JSpinner portSpinner = new JSpinner(portModel);
+
+    final int width1 = hostLabel.getPreferredSize().width;
+    final int width2 = portSpinner.getPreferredSize().width;
+    final int height1 = hostField.getPreferredSize().height;
+    final int[] rows = { height1, height1 };
+    final int[] cols = { width1, width2, TableLayout.FILL };
+
+    final JPanel hostPanel = new JPanel();
+    hostPanel.setLayout(new TableLayout(rows, cols));
+
+    hostPanel.add(hostLabel, new Rectangle(0, 0, 1, 1));
+    hostPanel.add(hostField, new Rectangle(1, 0, 2, 1));
+    hostPanel.add(portLabel, new Rectangle(0, 1, 1, 1));
+    hostPanel.add(portSpinner, new Rectangle(1, 1, 1, 1));
+    return hostPanel;
+  }
+
+  private void setupCredentials() {
+    JLabel usernameLabel = new JLabel("Username:");
+    JLabel passwordLabel = new JLabel("Password:");
+
+    usernameField = new JTextField();
+    passwordField = new JPasswordField();
+
+    usernameField.addKeyListener(this);
+    passwordField.addKeyListener(this);
+
+    final int width1 = usernameLabel.getPreferredSize().width;
+    final int width2 = passwordLabel.getPreferredSize().width;
+    final int width3 = Math.max(width1, width2);
+    final int height1 = usernameField.getPreferredSize().height;
+
+    final int[] rows = { height1, height1 };
+    final int[] cols = { width3, TableLayout.FILL };
+
+    credentials = new JGroup("Server requires authentication", false);
+    credentials.addActionListener(this);
+    credentials.setInnerLayout(new TableLayout(rows, cols));
+
+    credentials.addInner(usernameLabel, new Rectangle(0, 0, 1, 1));
+    credentials.addInner(passwordLabel, new Rectangle(0, 1, 1, 1));
+    credentials.addInner(usernameField, new Rectangle(1, 0, 1, 1));
+    credentials.addInner(passwordField, new Rectangle(1, 1, 1, 1));
+
+    credentials.setSelected(false);
+  }
+
+  public ProxySettings getUpdatedSettings() {
+    proxySettings.setHost(hostField.getText());
+    proxySettings.setPort(portModel.getNumber().intValue());
+
+    if (credentials.isSelected()) {
+      proxySettings.setUsername(usernameField.getText());
```

Comment:
I think it might be better to check if usernameField.getText().trim().length() > 0 and then call setUsername

Reply:
I guess when you do it that way the user might think that they *need* to enter a username and password.